spa/src/interface/cpu.rs

```diff
@@ -2,27 +2,149 @@
 // SPDX-FileCopyrightText: Copyright (c) 2025 Asymptotic Inc.
 // SPDX-FileCopyrightText: Copyright (c) 2025 Sanchayan Maity
 
+use super::plugin::Interface;
+use bitflags::bitflags;
+use pipewire_native_macros::EnumU32;
 use std::{any::Any, pin::Pin};
 
-use super::plugin::Interface;
+bitflags! {
+    #[repr(C)]
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, EnumU32)]
+    pub struct X86CpuFlags: u32 {
+        const MMX            = (1<<0);	/* standard MMX */
+        const MMXEXT         = (1<<1);	/* SSE integer or AMD MMX ext */
+        const AMD_3DNOW      = (1<<2);	/* AMD 3DNOW */
+        const SSE            = (1<<3);	/* SSE */
+        const SSE2           = (1<<4);	/* SSE2 */
+        const AMD_3DNOWEXT   = (1<<5);	/* AMD 3DNowExt */
+        const SSE3           = (1<<6);	/* Prescott SSE3 */
+        const SSSE3          = (1<<7);	/* Conroe SSSE3 */
+        const SSE41          = (1<<8);	/* Penryn SSE4.1 */
+        const SSE42          = (1<<9);	/* Nehalem SSE4.2 */
+        const AESNI          = (1<<10);	/* Advanced Encryption Standard */
+        const AVX            = (1<<11);	/* AVX */
+        const XOP            = (1<<12);	/* Bulldozer XOP */
+        const FMA4           = (1<<13);	/* Bulldozer FMA4 */
+        const CMOV           = (1<<14);	/* supports cmov */
+        const AVX2           = (1<<15);	/* AVX2 */
+        const FMA3           = (1<<16);	/* Haswell FMA3 */
+        const BMI1           = (1<<17);	/* Bit Manipulation Instruction Set 1 */
+        const BMI2           = (1<<18);	/* Bit Manipulation Instruction Set 2 */
+        const AVX512         = (1<<19);	/* AVX-512 */
+        const SLOW_UNALIGNED = (1<<20);	/* unaligned loads/stores are slow */
+        const _              = !0;      /* The source may set any bits */
+    }
+}
+
+bitflags! {
+    #[repr(C)]
+    #[derive(Debug, Clone, Copy, PartialEq, Eq, EnumU32)]
+    pub struct PPCCpuFlags: u32 {
```

Comment:
For consistency, i think the flags should have camelcase even for acronym arch names (we use Cpu not CPU), so it should be Arm not ARM).